### ISSUE TITLE
Equations having redundant non zero variables are not simplified

### ISSUE THREAD

**Issue by shantypat** [sympy/sympy#21108]
`import sympy as sp
sp.var('a b c',nonzero=True)
eq = sp.Eq(a*b-a*c,0)
print(eq)
print(eq.simplify())`

output : 
Eq(a*b - a*c, 0)
Eq(a*(b - c), 0)

**Comment by variable-jain** [comment]
Hi, I would like to work on this issue. Can you please elaborate your problem. 
What are you expecting the output here?

**Comment by smichr** [comment]
If non of the symbols is 0 then factors which are known to be non-zero (in this case `a`) can be eliminated to give `Eq(b - c, 0)`. If `b` were odd and `c` were even then `False` could be returned since if all factors are non-zero and finite then the result cannot be zero.

**Comment by oscarbenjamin** [comment]
I'm not sure that this makes sense as a simplification in all situations. Would it not be surprising in some cases that simplifying an equation leads to symbols being removed?

**Comment by rursprung** [comment]
i have another example where this simplification would be useful:

```python
import sympy as sym

a = sym.symbols('a', real=True, positive=True, nonzero=True)
b = sym.symbols('b', real=True)

display(sym.Eq(2*b,0).simplify())
display(sym.Eq(2*a*b,0).simplify())
```

currently (with sympy 1.12) this prints `b=0` for the former (correct, it correctly divided both sides by `2`) but `2*a*b=0` for the latter (rather than also simplifying it to `b=0`). as `a` is defined as being nonzero it is safe to divide by `2*a` without losing anything in this formula.
i'm also not sure why it doesn't simplifies the factor `2` away in the latter example since the only way for `2*a=0` to be the case is if `a=0` (which, in turn, it cannot be).

when you run `sym.solve(sym.Eq(2*a*b,0), b)` it correctly evaluates to `b=0`.

**Comment by asmeurer** [comment]
> when you run sym.solve(sym.Eq(2*a*b,0), b) it correctly evaluates to b=0.

In general, isn't `solve` really the simplification function for equations, in the sense being discussed here?

**Comment by rursprung** [comment]
> > when you run sym.solve(sym.Eq(2_a_b,0), b) it correctly evaluates to b=0.
> 
> In general, isn't `solve` really the simplification function for equations, in the sense being discussed here?

this was an (over?)simplified example which reduces to a single variable. you can imagine the same where you do not wish to reduce re-order the equation to a single variable using `solve` and instead want to proceed with it in other directions (e.g. running a laplace transform on it when you're working on control engineering).

**Comment by oscarbenjamin** [comment]
I don't think that `simplify` is the right function for this. A function for doing this needs to be told (in the way that `solve` is) which symbols are of interest.
